ergo-rest/src/api/node.rs

```diff
@@ -46,18 +47,437 @@ pub async fn get_info(node: NodeConf) -> Result<NodeInfo, NodeError> {
 }
 
 /// GET on /peers/all endpoint
-pub async fn get_peers_all(_node: NodeConf) -> Result<Vec<PeerInfo>, NodeError> {
-    todo!()
+async fn get_peers_all(node: NodeConf) -> Result<Vec<PeerInfo>, NodeError> {
+    #[allow(clippy::unwrap_used)]
+    let url = node.addr.as_http_url().join("peers/all").unwrap();
+    let client = build_client(&node)?;
+    let rb = client.get(url);
+    let response = set_req_headers(rb, node).send().await?;
+    Ok(response.json::<Vec<PeerInfo>>().await?)
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+/// Given a list of seed nodes, recursively determine a Vec of all known peer nodes. Note that there
+/// are `println` calls below that are commented out. Uncommenting them will show the requests that
+/// are made, especially when made in parallel.
+pub async fn peer_discovery(
+    seeds: NonEmptyVec<Url>,
+    max_parallel_requests: BoundedU16<1, { u16::MAX }>,
+    timeout: Duration,
```

Comment:
Initially, I was planning for `timeout` to be an "overall" timeout for the whole peer discovery run. In case there are way too many peers to discover them all in a reasonable amount of time. 
We might get away without it now since the number of active nodes is quite low but I'd rather have it implemented now.
What do you think?

Reply:
Yeah that's a great idea. Perhaps the timeout can be a 'best-effort' one: once timeout is reached, return the peers discovered up to that point in time. Sounds ok?

Reply:
Exactly!

Reply:
I've implemented it, but it requires us to vendor another crate. To implement timeout we need an async way to sleep. Tokio already has this built-in but for WASM there isn't a standard to rely on.

There are 2 possible crates that do this: `futures-timer` and `wasm-timer`. Both of them in current form break the `webpack` build process. I couldn't figure out how to fix it with `futures-timer`, but tracked it down with `wasm-timer`. Basically, there's an out of date dependency that is at fault. Now this crate hasn't been updated since 2020, hence my vendoring. Hope this is ok.